app/controllers/users_controller.rb

```diff
@@ -1,13 +1,21 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action :set_user, only: %i[show liked followers following discover feed]
+  before_action :correct_user_feed, only: %i[feed]
+  before_action :correct_user_discover, only: %i[discover]
+
```

Comment:
I combined user feed and discover because I thought they served the same function, but I think separating them is good for readability. 👍